### PR TITLE
402 refactor merge modules losses distance and metrics to one

### LINKED ISSUE #402: [Refactor] Merge modules `losses`, `distance`, and `metrics` to one
# Merge modules `losses`, `distance`, and `metrics` to one.
## Modules to be merged
- kale.utils.distance
- kale.predict.losses
- kale.evaluate.metrics
## Impact on code structure
The functions in these modules performed similarly, like calculating loss, accuracy, etc. It could be confusing when people try to use them or contribute to the library.
The import parts in some files may be changed.

### PR DESCRIPTION
**Pull request by shuo-zhou**
Fixes #402.

### Description
Merge modules `losses`,  `distance`, and `metrics` to one under `evaluate`.

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [ ] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] In-line docstrings updated.
- [x] [Source for documentation at `docs`](https://github.com/pykale/pykale/tree/main/docs/source) manually updated for new API.

### CODE REVIEW

File: tests/evaluate/test_metrics.py
Comment (jiayang-zhang) [inline tests/evaluate/test_metrics.py:1]: Yes, I can confirm the code added to this file -- as it is the exact code from `tests/utils/test_distance.py` (which is now deleted).

File: tests/utils/test_distance.py
Comment (jiayang-zhang) [inline tests/utils/test_distance.py:1]: Yes, I can confirm the file deletion -- as the exact code has been transferred to `tests/evaluate/test_metrics.py`.

**Review by jiayang-zhang — COMMENTED** [review]
Review completed.

File: examples/fewshot_protonet/README.md
[the remark this answers is not in the document]
Comment (wenruifan) [reply examples/fewshot_protonet/README.md:1]: I can confirm it. It is a typo. It should be `kale.predict.losses.protonet_loss`.

File: tests/evaluate/test_metrics.py [hunk 1]
```diff
@@ -78,3 +84,36 @@ def test_proto_loss():
     prototypes = feature_support.mean(dim=1)
     dists = loss_fn.euclidean_dist_for_tensor_group(prototypes, feature_query)
     assert dists.shape == (num_classes, num_classes * num_query_samples)
+
```
Comment (alanbijuthomas) [inline tests/evaluate/test_metrics.py:87]: Rename test_proto_loss to test_protonet_loss.

**Review by alanbijuthomas — CHANGES_REQUESTED** [review]
Only a minor change required - renaming of a test function.

**Pull request merged**